src/nvim/buffer.c

```diff
@@ -97,6 +97,13 @@ typedef enum {
   kBffInitChangedtick = 2,
 } BufFreeFlags;
 
+/// Allocate new buf structure and initialize it.
+buf_T *buf_alloc(void) {
+  buf_T *buf = xcalloc(1, sizeof(buf_T));
+  buf->b_changed_notified = true;
```

Comment:
This should just be below `buf->b_signcols_max = -1;` in `buflist_new` ( a bit weirdly named function but it is already the only entry point for allocating buffers.)